### ISSUE TITLE
Support different levels of indentation for MemberExpression with indent at indent level 0

### ISSUE THREAD

**Issue by w0rp** [eslint/eslint#8485]
**Tell us about your environment**

* **ESLint Version:** v4.0.0-alpha.0
* **Node Version:** v6.4.0
* **npm Version:** yarn v0.21.3

I have some AngularJS code like so:

```javascript
angular.module('some.module.string', [])
.component('componentName', {
  controller: function() {
    foo()
      .bar()
  },
})
```

I would like the code in the global scope _not to be_ up-indented, but I would like method chains inside of my functions _to be_ up-indented. As far as I'm aware, it's not possible to configure `eslint` to enforce this. For now, I am just setting `MemberExpression: "off"` for the `indent` rule.

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. Could you be more specific about how this would work? For example,
 if you're proposing a new option on the rule, what should the option do specifically? (Answering the questions in the [rule change proposal template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md) would be useful too.)

**Comment by w0rp** [comment]
Sure, I'll give that a go. I hope this will be informative.

Thanks to everyone for `eslint`, by the way. I swear by it as one of the most powerful development tools in existence.

**What rule do you want to change?**

`indent`

**Does this change cause the rule to produce more or fewer warnings?**

Possibly the same amount. It depends on the codebase.

**How will the change be implemented? (New option, new default behavior, etc.)?**

`MemberExpression` could accept an object, say with two keys `indentSize` and `globalScopeIndentSize`. `globalScopeIndentSize` would allow a different size of indentation to be set for method chains in the global scope.

(The implementation doesn't really matter, just the ability to configure the right behaviour.)

**Please provide some example code that this change will affect:**

The following code will be permitted:

```javascript
angular.module('some.module.string', [])
.component('componentName', {
  controller: function() {
    foo()
      .bar()
  },
})
```

Rather than the following being suggested:

```javascript
angular.module('some.module.string', [])
  .component('componentName', {
    controller: function() {
      foo()
        .bar()
    },
  })
```

Or the following:

```javascript
angular.module('some.module.string', [])
.component('componentName', {
  controller: function() {
    foo()
    .bar()
  },
})
```

**What does the rule currently do for this code?**

The `indent` rule can be configured to permit one of the two examples above, but not the first one, without turning the `indent` checks off completely for member chains.

**What will the rule do after it's changed?**

The rule will permit a particular style which can be useful for AngularJS codebases.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
